### PR TITLE
Handle panics in functions executed as separate goroutines

### PR DESCRIPTION
**Pull request by inventa-consulting**
Panics (Go exceptions) that occurred in the 3 functions called as separate goroutines (threads) were not handled and caused Monstache to crash. Defer functions with error handling were added inside the 3 functions

### CODE REVIEW

File: docker/plugin/digitile-monstache-plugin.go [hunk 1]
```diff
@@ -79,7 +80,7 @@ func ProcessDocument(input *monstachemap.MapperPluginInput) (output *monstachema
 
 	defer func() {
 		if recoveredError := recover(); recoveredError != nil {
-			err = fmt.Errorf("Syncing asset with ID %s failed with an unknow error: %s",
+			err = fmt.Errorf("Syncing asset with ID %s failed with an unknown error: %s",
 				assetId, recoveredError)
 		}
 	}()
```
Comment (inventa-consulting) [inline docker/plugin/digitile-monstache-plugin.go:86]: This defer function only handle panics within the function that contains it and not in goroutines called from this function.

File: docker/plugin/digitile-monstache-plugin.go [hunk 2]
```diff
@@ -183,6 +184,13 @@ func AddFileTypeFields(input *monstachemap.MapperPluginInput, document map[strin
 
 	defer wg.Done()
 
+	defer func() {
+		if recoveredError := recover(); recoveredError != nil {
+			lookupErrors <- fmt.Sprintf("Adding filetype details failed with an unknown error: %s\n%s",
+				recoveredError, string(debug.Stack()))
+		}
+	}()
+
```
Comment (inventa-consulting) [inline docker/plugin/digitile-monstache-plugin.go:193]: Separate defer function for handling panics when adding filetype fields

File: docker/plugin/digitile-monstache-plugin.go [hunk 3]
```diff
@@ -272,6 +280,13 @@ func AddTagNames(input *monstachemap.MapperPluginInput, document map[string]inte
 
 	defer wg.Done()
 
+	defer func() {
+		if recoveredError := recover(); recoveredError != nil {
+			lookupErrors <- fmt.Sprintf("Adding tag details failed with an unknown error: %s\n%s",
+				recoveredError, string(debug.Stack()))
+		}
+	}()
```
Comment (inventa-consulting) [inline docker/plugin/digitile-monstache-plugin.go:288]: Separate defer function for handling panics when adding tag names

File: docker/plugin/digitile-monstache-plugin.go [hunk 4]
```diff
@@ -362,6 +377,13 @@ func AddUserDetails(input *monstachemap.MapperPluginInput, document map[string]i
 
 	defer wg.Done()
 
+	defer func() {
+		if recoveredError := recover(); recoveredError != nil {
+			lookupErrors <- fmt.Sprintf("Adding user details failed with an unknown error: %s\n%s",
+				recoveredError, string(debug.Stack()))
+		}
+	}()
```
Comment (inventa-consulting) [inline docker/plugin/digitile-monstache-plugin.go:385]: Separate defer function for handling panics when adding user details

**Review by mhillebrand-digitile — APPROVED** [review]

**Review by jtopel — APPROVED** [review]

**Pull request merged**